cli/flox/src/commands/publish.rs

```diff
@@ -184,121 +216,92 @@ mod tests {
                     key_file: key_args.clone(),
                 }),
             },
             TestCase {
-                name: "config when args None",
+                name: "ingress_uri when args None",
                 config: Some(PublishConfig {
-                    store_url: Some(url_config.clone()),
                     signing_key: Some(key_config.clone()),
                 }),
-                args: None,
+                args: CacheArgs {
+                    store_url: None,
+                    signing_private_key: None,
+                },
+                ingress_uri: Some(url_response.clone()),
                 expected: Some(NixCopyCache {
-                    url: url_config.clone(),
+                    url: url_response.clone(),
                     key_file: key_config.clone(),
                 }),
             },
             TestCase {
                 name: "args when both Some",
                 config: Some(PublishConfig {
-                    store_url: Some(url_config.clone()),
                     signing_key: Some(key_config.clone()),
                 }),
-                args: Some(CacheArgs {
+                args: CacheArgs {
                     store_url: Some(url_args.clone()),
-                    signing_key: Some(key_args.clone()),
-                }),
+                    signing_private_key: Some(key_args.clone()),
+                },
+                ingress_uri: Some(url_response.clone()),
                 expected: Some(NixCopyCache {
                     url: url_args.clone(),
                     key_file: key_args.clone(),
                 }),
             },
             TestCase {
-                name: "mix of url from config and key from args",
-                config: Some(PublishConfig {
-                    store_url: Some(url_config.clone()),
-                    signing_key: None,
-                }),
-                args: Some(CacheArgs {
+                name: "mix of url from response and key from args",
+                config: Some(PublishConfig { signing_key: None }),
+                args: CacheArgs {
                     store_url: None,
-                    signing_key: Some(key_args.clone()),
-                }),
+                    signing_private_key: Some(key_args.clone()),
+                },
+                ingress_uri: Some(url_response.clone()),
                 expected: Some(NixCopyCache {
-                    url: url_config.clone(),
+                    url: url_response.clone(),
                     key_file: key_args.clone(),
                 }),
             },
         ];
 
         for tc in test_cases {
             assert_eq!(
-                merge_cache_options(tc.config, tc.args).unwrap(),
+                merge_cache_options(tc.config, tc.args, tc.ingress_uri).unwrap(),
                 tc.expected,
                 "test case: {}",
                 tc.name
             );
         }
     }
-
-    #[test]
-    fn test_merge_cache_options_error() {
```

Comment:
followup question: how come the test for the error cases didn't get updated? Do we need to change the error message?